From 796439b636c2710c00a9aeadb3099e5e47261035 Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Thu, 30 Apr 2020 14:15:52 +0800
Subject: [PATCH] Let installd and dumpstate run as system user

Our customer want all the processes of a container to run
as non-root user. So let installd and dumpstate run as system user
and grant the necessary capabilities to it.

Change-Id: Idddef2b92a8b6bc01e044b4e819443d27b454ab1
Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 cmds/dumpstate/dumpstate.rc | 2 ++
 cmds/installd/installd.rc   | 1 +
 2 files changed, 3 insertions(+)

diff --git a/cmds/dumpstate/dumpstate.rc b/cmds/dumpstate/dumpstate.rc
index 2e7257473..8e8308bb9 100644
--- a/cmds/dumpstate/dumpstate.rc
+++ b/cmds/dumpstate/dumpstate.rc
@@ -5,6 +5,7 @@ on boot
 
 service dumpstate /system/bin/dumpstate -s
     class main
+    capabilities DAC_OVERRIDE DAC_READ_SEARCH SETUID SETGID SYS_RESOURCE KILL NET_RAW NET_ADMIN CHOWN FOWNER FSETID SYSLOG SYS_PTRACE
     socket dumpstate stream 0660 shell log
     disabled
     oneshot
@@ -15,5 +16,6 @@ service dumpstatez /system/bin/dumpstate -S -d -z \
         -o /data/user_de/0/com.android.shell/files/bugreports/bugreport
     socket dumpstate stream 0660 shell log
     class main
+    capabilities DAC_OVERRIDE DAC_READ_SEARCH SETUID SETGID SYS_RESOURCE KILL NET_RAW NET_ADMIN CHOWN FOWNER FSETID SYSLOG SYS_PTRACE
     disabled
     oneshot
diff --git a/cmds/installd/installd.rc b/cmds/installd/installd.rc
index 240aa495b..9de2897e5 100644
--- a/cmds/installd/installd.rc
+++ b/cmds/installd/installd.rc
@@ -1,6 +1,7 @@
 
 service installd /system/bin/installd
     class main
+    capabilities DAC_OVERRIDE DAC_READ_SEARCH CHOWN FOWNER FSETID SETGID SETUID SYS_ADMIN
 
 on early-boot
     mkdir /config/sdcardfs/extensions/1055
-- 
2.20.1

